examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -313,6 +317,32 @@ on_event(struct pt_regs* ctx) {
   event->stack_status = STACK_STATUS_ERROR;
   event->error_code = ERROR_NONE;
 
+  struct task_struct const *const task = (struct task_struct *)bpf_get_current_task();
+
+  // Get raw native user stack
+  struct pt_regs user_regs = *ctx;
+  if (!user_mode(&user_regs)) {
+    // The third argument is equivalent to `task_pt_regs(task)`
+    bpf_probe_read_kernel(
+        &user_regs, sizeof(user_regs),
+        (struct pt_regs *)((unsigned long)(task->stack) + THREAD_SIZE -
+                           TOP_OF_KERNEL_STACK_PADDING) - 1);
+  }
+
+  event->user_sp = user_regs.sp;
+  event->user_ip = user_regs.ip;
+
+  // Copy one page at the time - if one fails we don't want to lose the others
+  int i;
+  for (i = 0; i < __USER_STACKS_PAGES__; ++i) {
```

Comment:
`sizeof(event->raw_user_stack) / PAGE_SIZE`?

Reply:
Why?

Reply:
The loop writes to the array so it should be bounded on the array size. The fact that the array is also dependent on `__USER_STACKS_PAGES__` is something else.